examples/tensorrt_llm/common/base_engine.py

```diff
@@ -368,3 +406,102 @@ async def async_llm_wrapper():
+            )
+        logger.debug(
+            f"Received response from prefill worker: {ctx_responses[0].data()}"
+        )
+        ctx_response_obj = TRTLLMWorkerResponse.model_validate_json(
+            ctx_responses[0].data()
+        )
+        ctx_response_obj.outputs = [
+            TRTLLMWorkerResponseOutput(**ctx_response_obj.outputs[0])
+        ]
+        assert ctx_response_obj.outputs[0].disaggregated_params is not None
+
+        return ctx_response_obj
+
+    async def generate(self, request: TRTLLMWorkerRequest):
+        if self._llm_engine is None:
+            raise RuntimeError("Engine not initialized")
+
+        if self._error_queue and self._error_queue.qsize() > 0:
+            error = self._error_queue.get()
+            raise error
+
+        self._ongoing_request_count += 1
+
+        try:
+            worker_inputs = request.tokens.tokens
+
+            disaggregated_params = (
+                DisaggregatedTypeConverter.to_llm_disaggregated_params(
+                    request.disaggregated_params
+                )
+            )
+
+            if self._remote_prefill and self._server_type == ServerType.GEN:
+                ctx_response_obj = await self._get_remote_prefill_response(request)
+
+                worker_inputs = ctx_response_obj.prompt_token_ids
+                disaggregated_params = (
+                    DisaggregatedTypeConverter.to_llm_disaggregated_params(
+                        DisaggregatedParams(
+                            **ctx_response_obj.outputs[0].disaggregated_params
+                        )
+                    )
+                )
+                disaggregated_params.request_type = "generation_only"
+
+            logger.debug(
+                f"Worker inputs: {worker_inputs}, disaggregated params: {disaggregated_params}"
+            )
+
+            sampling_params = get_sampling_params(request.sampling_params)
+            async for response in self._llm_engine.generate_async(
+                inputs=worker_inputs,
+                sampling_params=sampling_params,
+                disaggregated_params=disaggregated_params,
+                streaming=(self._server_type == ServerType.GEN),
+            ):
+                # Encode the disaggregated params to OAI format so
+                # it can be sent over the network.
+                response.outputs[
+                    0
+                ].disaggregated_params = DisaggregatedTypeConverter.to_oai_disaggregated_params(
+                    response.outputs[0].disaggregated_params
+                )
+                yield TRTLLMWorkerResponse(
+                    request_id=request.id,
+                    prompt_token_ids=response.prompt_token_ids,
+                    outputs=[asdict(response.outputs[0])],
+                    finished=response.finished,
+                ).model_dump_json(exclude_unset=True)
+
+        except CppExecutorError:
+            signal.raise_signal(signal.SIGINT)
```

Comment:
should we log here? why sig int in this case? is this fatal?

Reply:
Not sure. Brought this code from the FastAPI implementation of trtllm-serve. May be it hints on an irrecoverable error.